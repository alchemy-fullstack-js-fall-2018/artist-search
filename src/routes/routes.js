import Search from '../components/search/Search';
import ArtistDetail from '../components/artist/ArtistDetail';
import Songs from '../components/song/Songs';

export const ROUTES = {
  SEARCH: {
    path: '/search',
    Component: Search,
    linkTo: () => '/search'
  },
  ARTIST: {
    path: '/artists/:id',
    Component: ArtistDetail,
    linkTo: id => `/artists/${id}`
  },
  SONG: {
    path:'/lyrics/:artist/:song',
    Component: Songs,
    linkTo: (artist, song) => `/lyrics/${artist}/${song}`
  }
};

