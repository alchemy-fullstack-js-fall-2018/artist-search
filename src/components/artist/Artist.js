import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';


export default function Arist ({ name, description, id }) {

  return (
    <div>
      <h3>{name} - {description} <Link to={ROUTES.ARTIST.linkTo(id)}>Get Details: </Link> </h3>
    </div>
  );
}
