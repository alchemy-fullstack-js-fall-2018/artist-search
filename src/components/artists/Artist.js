import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';
import PropTypes from 'prop-types';

const Artist = ({ name, description, id }) => {
  return (
    <div>
      <h3>
        {name} - {description}
      </h3>
      <Link to={ROUTES.ARTIST.linkTo(id)}>Get Works by {name}.</Link>
    </div>
  );
};

Artist.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string,
  id: PropTypes.string.isRequired
};

export default Artist;
