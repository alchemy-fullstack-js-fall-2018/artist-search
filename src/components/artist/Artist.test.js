import React from 'react';
import Artist from '../artist/Artist';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';

describe('Artist', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Artist id="111" name="a cool band" description="gross" />);
  });


  it('renders a Artist', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays the name and description', () => {
    const wrapperWithRouter = shallow(
      <MemoryRouter>
        <Artist id="111" name="a cool band" description="gross" />
      </MemoryRouter>

    );
    expect(wrapperWithRouter.html()).toContain('gross');
    expect(wrapperWithRouter.html()).toContain('a cool band');
  });

  it('links to a artist detail', () => {
    expect(wrapper.find('Link').props().to).toEqual(ROUTES.ARTIST.linkTo(111));
  });
});
