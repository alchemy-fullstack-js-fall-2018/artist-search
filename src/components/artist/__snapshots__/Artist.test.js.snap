// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist renders a Artist 1`] = `
<div>
  <h3>
    a cool band
     - 
    gross
  </h3>
  <Link
    replace={false}
    to="/artists/111"
  >
    Get Details
  </Link>
</div>
`;
